server/channels/app/post_metadata.go

```diff
@@ -175,6 +175,8 @@ func (a *App) getEmbedsAndImages(c request.CTX, post *model.Post, isNewPost bool
 		post.Metadata.Embeds = []*model.PostEmbed{}
 	}
 
+	removeEmbeddedPostsFromMetadata(post)
```

Comment:
Is this potentially something that could be removed earlier? I went into more detail in Jira, and I want to avoid splitting the conversation here

https://mattermost.atlassian.net/browse/MM-59078?focusedCommentId=168285

Reply:
Makes sense. Done.